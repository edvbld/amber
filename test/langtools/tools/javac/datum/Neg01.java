/*
 * @test /nodynamiccopyright/
 * @summary smoke negative test for datum classes
 * @compile/fail/ref=Neg01.out -XDrawDiagnostics Neg01.java
 */

public class Neg01 {
    static abstract __datum Sup1(int x, int y) { }

    __datum Bad1(int x, int y, int z) extends Sup1(x, y, z) { } //too many super args

    __datum Bad2(int x, int z) extends Sup1(x) { } //too few super args

    __datum Bad3(int x, int z) extends Sup1(x, z) { } //name mismatch

    __datum Bad4(int x, double y) extends Sup1(x, y) { } //type mismatch

    __datum Bad5(int x, int y) extends Sup1(x, y) {
        Bad5(int x, int y) { super(x, y); } //error: explicit constructor and super header
    }

    static class Sup2 { }

    __datum Bad6(int x, int y) extends Object(x) { } //bad super header

    __datum Bad7(int x, int y) extends Sup2 { } //non-datum superclass

    static __datum Sup3(int x, int y) { }

    __datum Bad7(int x, int y) extends Sup2 { } //non-abstract datum superclass

    __datum Test(int x) {
        Test(int x, int y) {
            default(); //too few
            default(x); //ok
            default(x, y); //too many
        }

        void test() {
           default(); //error - not in a constructor
        }
    }
}
