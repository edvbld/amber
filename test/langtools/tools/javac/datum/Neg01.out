Neg01.java:30:5: compiler.err.already.defined: kindname.class, Neg01.Bad7, kindname.class, Neg01
Neg01.java:10:5: compiler.err.cant.apply.symbol: kindname.constructor, Sup1, int,int, int,int,int, kindname.class, Neg01.Sup1, (compiler.misc.arg.length.mismatch)
Neg01.java:12:5: compiler.err.cant.apply.symbol: kindname.constructor, Sup1, int,int, int, kindname.class, Neg01.Sup1, (compiler.misc.arg.length.mismatch)
Neg01.java:14:40: compiler.err.cant.extend.datum: (compiler.misc.super.field.mismatch: int, y, int, z)
Neg01.java:16:43: compiler.err.cant.extend.datum: (compiler.misc.super.field.mismatch: int, y, double, y)
Neg01.java:16:5: compiler.err.cant.apply.symbol: kindname.constructor, Sup1, int,int, int,double, kindname.class, Neg01.Sup1, (compiler.misc.no.conforming.assignment.exists: (compiler.misc.possible.loss.of.precision: double, int))
Neg01.java:18:40: compiler.err.cant.extend.datum: (compiler.misc.bad.super.fields)
Neg01.java:24:40: compiler.err.cant.extend.datum: (compiler.misc.bad.datum.super)
Neg01.java:24:5: compiler.err.cant.apply.symbol: kindname.constructor, Object, compiler.misc.no.args, int, kindname.class, java.lang.Object, (compiler.misc.arg.length.mismatch)
Neg01.java:26:40: compiler.err.cant.extend.datum: (compiler.misc.bad.datum.super)
Neg01.java:34:20: compiler.err.cant.apply.symbol: kindname.constructor, Test, int, compiler.misc.no.args, kindname.class, Neg01.Test, (compiler.misc.arg.length.mismatch)
Neg01.java:36:20: compiler.err.cant.apply.symbol: kindname.constructor, Test, int, int,int, kindname.class, Neg01.Test, (compiler.misc.arg.length.mismatch)
Neg01.java:40:12: compiler.err.cant.resolve.location.args: kindname.method, default, , , (compiler.misc.location: kindname.class, Neg01.Test, null)
13 errors
